cmake/bintools/gnu/target_strip.cmake

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Construct a commandline suitable for calling the toolchain binary tools
+#  version of strip.
+#
+#  Usage:
+#    bintools_strip(
+#      RESULT_CMD_LIST    <List of commands to be executed, usually after build>
+#      RESULT_BYPROD_LIST <List of command output byproducts>
+#
+#      STRIP_ALL          <When present, remove relocation and symbol info>
+#      STRIP_DEBUG        <When present, remove debugging symbols and sections>
+#      STRIP_DWO          <When present, remove DWARF .dwo sections>
+#
+#      FILE_INPUT         <The input file>
+#      FILE_OUTPUT        <The output file>
+#    )
+macro(bintools_strip)
+  cmake_parse_arguments(
+    # Prefix of output variables
+    BINTOOLS_STRIP
+    # List of argument names without values, hence boolean
+    "STRIP_ALL;STRIP_DEBUG;STRIP_DWO"
+    # List of argument names with one value
+    "RESULT_CMD_LIST;RESULT_BYPROD_LIST;FILE_INPUT;FILE_OUTPUT"
+    # List of argument names with multible values
+    ""
+    # Parser input
+    ${ARGN}
+  )
+
+  if(NOT DEFINED BINTOOLS_STRIP_RESULT_CMD_LIST OR NOT DEFINED ${BINTOOLS_STRIP_RESULT_CMD_LIST})
+    message(FATAL_ERROR "RESULT_CMD_LIST is required.")
+  elseif(NOT DEFINED BINTOOLS_STRIP_FILE_INPUT OR NOT DEFINED BINTOOLS_STRIP_FILE_OUTPUT)
+    message(FATAL_ERROR "Both FILE_INPUT and FILE_OUTPUT are required.")
+  endif()
+
+  # Handle stripping
+  set(btSStrip "")
+  if(${BINTOOLS_STRIP_STRIP_ALL})
+    set(btSStrip "--strip-all")
```

Comment:
We use snake-case, not camel-case in CMake code.

Reply:
I see, my bad. I seem to have camel cased all in this PR.
Do you want all changed to snake case, or can we keep the camels for the binary tools?

Reply:
Change all to snake case.

Reply:
The camels have been removed and snakes inserted instead.